Make serverless function naming and Docker image naming consistent

SpecLad opened the pull request:
For function names, take the relative path, lowercase, join with dashes and replace underscores with dashes.

For image names, take the relative path, lowercase and join with dots.

In a couple cases, rename the function directory instead of fixing the config.

<!-- Raise an issue to propose your change (https://github.com/opencv/cvat/issues).
It helps to avoid duplication of efforts from multiple independent contributors.
Discuss your ideas with maintainers to be sure that changes will be approved and merged.
Read the [Contribution guide](https://opencv.github.io/cvat/docs/contributing/). -->

<!-- Provide a general summary of your changes in the Title above -->

### Motivation and context
<!-- Why is this change required? What problem does it solve? If it fixes an open
issue, please link to the issue here. Describe your changes in detail, add
screenshots. -->
All these inconsistencies are driving me batty. Derived names should be derived the same way everywhere in order to be predictable.

### How has this been tested?
<!-- Please describe in detail how you tested your changes.
Include details of your testing environment, and the tests you ran to
see how your change affects other areas of the code, etc. -->

### Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply.
If an item isn't applicable for some reason, then ~~explicitly strikethrough~~ the whole
line. If you don't do that, GitHub will show incorrect progress for the pull request.
If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I submit my changes into the `develop` branch
- ~~[ ] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file~~
- [x] I have updated the documentation accordingly
- ~~[ ] I have added tests to cover my changes~~
- ~~[ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- ~~[ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] ] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

SpecLad commented:
The idea is originally from #6129, but I fixed all functions this time, not just OpenVINO ones.

Here's the script I used to detect mismatches:

```python
import yaml

from pathlib import Path

for config_path in Path('serverless').glob("**/*.yaml"):
    config_rel = config_path.relative_to('serverless')

    assert config_rel.parent.name == 'nuclio'

    parts = list(config_rel.parents[1].parts)

    if parts[0] == 'pytorch':
        parts[0] = 'pth'
    elif parts[0] == 'tensorflow':
        parts[0] = 'tf'

    with config_path.open('rb') as config_file:
        config = yaml.safe_load(config_file)

    expected_name = '-'.join(part.replace('_', '-').lower() for part in parts)
    actual_name = config['metadata']['name']

    if actual_name != expected_name:
        print(f'{config_path}: name mismatch: expected {expected_name!r} actual {actual_name!r}')

    expected_image_name = 'cvat.' + '.'.join(part.lower() for part in parts)
    actual_image_name = config['spec']['build']['image']

    if actual_image_name != expected_image_name:
        print(f'{config_path}: image name mismatch: expected {expected_image_name!r} actual {actual_image_name!r}')
```

Probably should add something like this to CI, but no time for that right now.

bsekachev commented:
Quick look at our documentation shows that we still have some references using previous naming. 
For example:

<img width="471" alt="image" src="https://github.com/opencv/cvat/assets/40690378/22e4014f-1c5e-4311-a730-b24aeabbf419">

Also, If we rename SAM, we need to update our SAM article: https://www.cvat.ai/post/facebook-segment-anything-model-in-cvat (better to collaborate with Maria to know if we published it somewhere else).

And just thoughts, this patch will result our users have several deployed models with the same name in CVAT. Maybe it is not a big problem.

SpecLad commented:
> Quick look at our documentation shows that we still have some references using previous naming.

I believe those are all referring to the TensorFlow version (which hasn't changed).

> Also, If we rename SAM, we need to update our SAM article

Yeah, that's a fair point. I don't want to deal with it right now, so I just reverted the SAM changes. I might return to it later.

> And just thoughts, this patch will result our users have several deployed models with the same name in CVAT. Maybe it is not a big problem.

I don't think it's a big problem (especially given that I already updated a whole bunch of names in #6100), but just in case, I added a changelog entry describing the name changes.
